x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/async/AsyncTaskServiceTests.java

```diff
@@ -88,10 +89,16 @@ public String getFeatureDescription() {
     }
 
     public void testEnsuredAuthenticatedUserIsSame() throws IOException {
-        Authentication original = new Authentication(new User("test", "role"), new Authentication.RealmRef("realm", "file", "node"), null);
+        Authentication original = AuthenticationTestHelper.builder()
+            .user(new User("test", "role"))
+            .realmRef(new Authentication.RealmRef("realm", "file", "node"))
+            .build(false);
```

Comment:
The `false` argument here is to make sure the randomized Authentication is not run-as because that is how the Authentication is created in the current code (`null` for lookup realm). I suspect most of the tests would work with a run-as Authentication. But I chose to be conservative to reduce the complexity of this PR.